src/Http/Controllers/Character/WalletController.php

```diff
@@ -53,24 +56,73 @@ public function getJournal(int $character_id)
      */
     public function getJournalData(int $character_id)
     {
+        if (! request()->has('all_linked_characters'))
+            return response('required url parameter is missing!', 400);
 
-        $journal = $this->getCharacterWalletJournal($character_id, false);
+        if (request('all_linked_characters') === 'false')
+            $character_ids = collect($character_id);
+
+        $user_group = User::find($character_id)->group->users
+            ->filter(function ($user) {
+                return $user->name !== 'admin' && $user->id !== 1;
+            })
+            ->pluck('id');
+
+        if (request('all_linked_characters') === 'true')
+            $character_ids = $user_group;
+
+        $journal = $this->getCharacterWalletJournal($character_ids);
 
         return DataTables::of($journal)
             ->editColumn('ref_type', function ($row) {
 
-                return view('web::partials.journaltranstype', compact('row'))
-                    ->render();
+                return view('web::partials.journaltranstype', compact('row'));
             })
             ->editColumn('first_party_id', function ($row) {
 
-                return view('web::partials.journalfrom', compact('row'))
-                    ->render();
+                $character_id = $row->character_id;
+
+                if ($row->first_party->category === 'character') {
```

Comment:
actually, `first_party` may be null since `UniverseName` is only seed by frontend.
This can lead to property on null object exception